web/src/lib/components/LoginModal.svelte

```diff
@@ -1,31 +1,62 @@
+<!--
+    This is a store-based component to show a Login modal on the screen.
+
+    You can request to open the login modal from everywhere by writing
+    to the loginModalState:
+
+    import { loginModalState } from "$lib/stores";
+
+    loginModalState.set({
+       openRequested: true,
+       callbackFunc: function(){
+          console.log("Login correct, now let's do real stuff");
+          doSomethingUsefulAfterUserLoginCorrect();
+       }
+    });
+
+    - For the modal to open, you need to set `openRequested` to true.
+    - Also, you have to provide a callback function to be executed
+        when/if the login is correct. If you don't want any action
+        to be executed after login, just provide an empty function:
+
+        loginModalState.set({
+           openRequested: true,
+           callbackFunc: () => {}
+        });
+-->
 <svelte:options accessors />
 
 <script lang="ts">
     import Login from "$lib/components/Login.svelte";
+    import { loginModalState } from "$lib/stores";
+
+    let login: Login | null;
 
-    let login : Login | null;
-    let isModalOpen = false;
+    $: if ($loginModalState?.openRequested ?? false) {
+        show($loginModalState.callbackFunc);
+    }
 
     export function show(onLoginFunction) {
         if (typeof onLoginFunction === 'function') {
             onLogin = onLoginFunction;
         }
 
         login.startCheckingLogin();
-
-        isModalOpen = true;
     }
 
     export function hide() {
         login.stopCheckingLogin();
 
-        isModalOpen = false;
+        loginModalState.set({
```

Comment:
we shouldn't touch the store here - just a local variable that indicates whether the modal is open